### PR TITLE
add tiers datasource

### PR DESCRIPTION
**Pull request by taimoorgit**
## Issues

https://github.com/OpsLevel/team-platform/issues/289

## Tophatting

```
Changes to Outputs:
  + all_tiers = {
      + tiers = [
          + {
              + alias = "tier_1"
              + id    = "Z2lkOi8vb3BzbGV2ZWwvVGllci8zNDU3"
              + index = 1
              + name  = "Tier 1"
            },
          + {
              + alias = "tier_2"
              + id    = "Z2lkOi8vb3BzbGV2ZWwvVGllci8zNDU4"
              + index = 2
              + name  = "Tier 2"
            },
          + {
              + alias = "tier_3"
              + id    = "Z2lkOi8vb3BzbGV2ZWwvVGllci8zNDU5"
              + index = 3
              + name  = "Tier 3"
            },
          + {
              + alias = "tier_4"
              + id    = "Z2lkOi8vb3BzbGV2ZWwvVGllci8zNDYw"
              + index = 4
              + name  = "Tier 4"
            },
        ]
    }
  + tier_1    = {
      + alias  = "tier_1"
      + filter = {
          + field = "alias"
          + value = "tier_1"
        }
      + id     = "Z2lkOi8vb3BzbGV2ZWwvVGllci8zNDU3"
      + index  = 1
      + name   = "Tier 1"
    }

```

### CODE REVIEW

File: opslevel/datasource_opslevel_tiers_all.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+func (d *TierDataSourcesAll) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "List of all Tier data sources",
+
+		Attributes: map[string]schema.Attribute{
+			"tiers": schema.ListNestedAttribute{
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: tierAttributes(map[string]schema.Attribute{}),
```
Comment (davidbloss) [inline opslevel/datasource_opslevel_tiers_all.go:47]:
```suggestion
					Attributes: tierDatasourceSchemaAttrs,
```

File: opslevel/datasource_opslevel_tier.go [hunk 2]
```diff
@@ -52,32 +93,15 @@ func (d *TierDataSource) Schema(ctx context.Context, req datasource.SchemaReques
...
-			"index": schema.Int64Attribute{
-				MarkdownDescription: "The numerical representation of the tier.",
-				Computed:            true,
-			},
-			"name": schema.StringAttribute{
-				Description: "The display name of the tier.",
-				Computed:    true,
-			},
-		},
+		Attributes: tierAttributes(map[string]schema.Attribute{}),
```
Comment (davidbloss) [inline opslevel/datasource_opslevel_tier.go:96]:
```suggestion
		Attributes: tierDatasourceSchemaAttrs,
```
🤔 Looks like `tierAttributes()` can be removed

Comment (taimoorgit) [reply opslevel/datasource_opslevel_tier.go]: @davidbloss removed

**Review by davidbloss — APPROVED** [review]
LGTM! 🎉

**Pull request merged**